src/Plugin/tmgmt/Translator/ThebigwordTranslator.php

```diff
@@ -0,0 +1,872 @@
+    // Import given data using XLIFF converter. Specify that passed content is
+    // not a file.
+    return $xliff_converter->import($data, FALSE);
+  }
+
+  /**
+   * Fetches translations for job items of a given job.
+   *
+   * @param \Drupal\tmgmt\JobInterface $job
+   *   A job containing job items that translations will be fetched for.
+   * @param string $state
+   *   The state of the files.
+   *
+   * @return bool
+   *   Returns TRUE if there are error messages during the process of
+   *   retrieving translations. Otherwise FALSE.
+   *
+   * @throws \Drupal\tmgmt\TMGMTException
+   */
+  public function addTranslatedFilesToJob(JobInterface $job, $state) {
+    // Search for placeholder item.
+    $result = RemoteMapping::loadByLocalData($job->id());
+    $remote = reset($result);
+    $this->setTranslator($job->getTranslator());
+    $project_id = $remote->getRemoteIdentifier2();
+    $translated = 0;
+    $not_translated = 0;
+    $had_errors = FALSE;
+
+    try {
+      // Get the files of this job.
+      $files = $this->requestFileinfo($state, $job);
+      /** @var JobItemInterface $job_item */
+      foreach ($job->getItems() as $job_item) {
+        $mappings = RemoteMapping::loadByLocalData($job->id(), $job_item->id(), 'tmgmt_thebigword');
+        $mapping = reset($mappings);
+        $ids = $mapping->getRemoteData('files');
+        $ids = array_keys($ids);
+        $file_id = reset($ids);
+        if (isset($files[$file_id])) {
+          try {
+            $this->addFileDataToJob($job, $state, $file_id);
+          }
+          catch (TMGMTException $e) {
+            $this->sendFileError('RestartPoint01', $file_id, $e->getMessage(), NULL, $job_item);
+            $job->addMessage('Error fetching the job item: @job_item.', array('@job_item' => $job_item->label()), 'error');
+            $had_errors = TRUE;
+            $not_translated++;
+            continue;
+          }
+          $translated++;
+        }
+        else {
+          $not_translated++;
+        }
+      }
+    }
+    catch (TMGMTException $e) {
+      drupal_set_message('Could not pull translation resources.', array(), 'error');
+      return FALSE;
+    }
+    finally {
+      if ($had_errors) {
+        $this->confirmUpload($project_id, 'RestartPoint01');
+      }
+      return [
+        'translated' => $translated,
+        'untranslated' => $not_translated,
+      ];
+    }
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function requestJobItemsTranslation(array $job_items) {
+    /** @var JobItemInterface $job_item */
+    foreach ($job_items as $job_item) {
+      $this->setTranslator($job_item->getJob()->getTranslator());
+
+      try {
+        $job_item_id = $job_item->id();
+        $project_id = $this->newTranslationProject($job_item_id, $job_item_id, $job_item->getJob()->getSetting('required_by'), $job_item->getJob()->getSetting('quote_required'), $job_item->getJob()->getSetting('category'));
```

Comment:
the idea is to make one new project for a group of job items, so needs to be outside of the loop.